File: 정수내림차순으로배치하기/javascript/20221019/solution.test.js
import { split, sort, join } from './solution';

test('쪼개기', () => {
  expect(split(118372)).toStrictEqual(['1', '1', '8', '3', '7', '2']);
});

test('내림차순으로 정렬하기', () => {
  const array = ['1', '1', '8', '3', '7', '2'];
  expect(sort(array)).toStrictEqual([8, 7, 3, 2, 1, 1]);
});

test('하나의 정수로 합치기', () => {
  const array = [8, 7, 3, 2, 1, 1];
  expect(join(array)).toBe(873211);
});

File: programmers/level1/정수내림차순으로배치하기/java/20221019/test/SolutionTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SolutionTest {

    @Test
    void split() {
        Solution solution = new Solution();

        String[] numbers = {"1", "1", "8", "3", "7", "2"};

        assertArrayEquals(numbers, solution.split(118372));
    }

    @Test
    void sort() {
        Solution solution = new Solution();

        String[] numbers = {"1", "1", "8", "3", "7", "2"};
        String[] sortedNumbers = {"8", "7", "3", "2", "1", "1"};

        assertArrayEquals(sortedNumbers, solution.sort(numbers));
    }

    @Test
    void join() {
        Solution solution = new Solution();

        String[] sortedNumbers = {"8", "7", "3", "2", "1", "1"};

        assertEquals(873211, solution.joinToLong(sortedNumbers));
    }

}
